### ISSUE TITLE
easy way to get the time left?

### ISSUE THREAD

**Issue by droundy** [rusticata/x509-parser#9]
I'm just looking at this crate, and it looks like it will do what I want, but it also looks somewhat awkward to use for a PEM file.  My goal is just do decide if (and when) I need to renew a certificate through lets-encrypt, so I don't need *much* in the way of parsing.

I'm thinking a method on `X509Certificate` or `TbsCertificate` or `Validity` that looks like
```
  fn time_to_expiration(&self) -> Option<std::time::Duration> { ... }
```
would be very convenient to me and presumably to others.  I'd be happy to submit such a patch if you can tell me where you'd like the method to be.  My preference would be at the very top level, but I could imagine putting it in all three or just in `Validity`.

**Comment by chifflier** [comment]
I think the `Validity` structure can be extended with such functions (it seems to be the best place)
Functions could also be added to the certificate itself, so that checking would be simple

**Comment by chifflier** [comment]
BTW, if we use `Duration`, we have to be extra careful, several of its methods can panic (which is not acceptable IMHO, especially in a library crate)

**Comment by chifflier** [comment]
Merged #10, thanks
